lib/Conversion/StableHLOToTTIR/StableHLOToTTIRPatterns.cpp

```diff
@@ -264,6 +264,67 @@ class StableHLOToTTIRConstantOpConversionPattern
   }
 };
 
+class StableHLOToTTIRConvolutionOpConversionPattern
+    : public OpConversionPattern<mlir::stablehlo::ConvolutionOp> {
+  using OpConversionPattern<
+      mlir::stablehlo::ConvolutionOp>::OpConversionPattern;
+
+public:
+  LogicalResult
+  matchAndRewrite(mlir::stablehlo::ConvolutionOp srcOp,
+                  mlir::stablehlo::ConvolutionOp::Adaptor adaptor,
+                  ConversionPatternRewriter &rewriter) const override {
+
+    auto outputType = mlir::cast<RankedTensorType>(
+        getTypeConverter()->convertType(srcOp.getResult().getType()));
+
+    tensor::EmptyOp outputTensor = rewriter.create<tensor::EmptyOp>(
+        srcOp.getLoc(), outputType.getShape(), outputType.getElementType());
+
+    std::vector<int64_t> strides =
+        adaptor.getWindowStrides().value_or(ArrayRef<int64_t>({1, 1})).vec();
```

Comment:
Do you need to verify the input parameters are correct? You can use the following API to do that. 

```
// Check legality of the operation
LogicalResult err = checkBasicLegality(srcOp, adaptor, rewriter);
  if (failed(err)) {
    return err;
}
```

Reply:
I'm not really sure whether I do or not. Should I add this anyway?

Reply:
Stablehlo already performs some checks on the inputs, but if there are any TTIR or TTNN specific limitations on valid inputs, you can check for them and notify the user.